src/common/atlas/cluster.ts

```diff
@@ -3,23 +3,45 @@ import type { ApiClient } from "./apiClient.js";
 import { LogId } from "../logger.js";
 
 const DEFAULT_PORT = "27017";
+
+function extractProcessIds(connectionString: string): Array<string> {
+    if (!connectionString || connectionString === "") return [];
+
+    // Extract host:port pairs from connection string
+    const matches = connectionString.match(/^mongodb:\/\/([^/]+)/);
+    if (!matches) {
+        return [];
+    }
+
+    // matches[1] gives us the host:port pairs
+    const hostsString = matches[1];
+    const hosts = hostsString?.split(",") ?? [];
+
+    return hosts?.map((host) => {
```

Comment:
hosts can't be null or undefined, because of the `??` branch.

```suggestion
    return hosts.map((host) => {
```